pkg/catalog/catalog.go

```diff
@@ -1842,13 +1832,59 @@ func (c *Catalog) writeUncommittedLocal(ctx context.Context, repository *gravele
 	if itr.Err() != nil {
 		return nil, false, itr.Err()
 	}
-	err = pw.WriteStop()
+
+	// Finished reading all staging area - no continuation token
+	hasData := count > 0
+	return nil, hasData, nil
+}
+
+func (c *Catalog) writeTokensLocal(ctx context.Context, repository *graveler.RepositoryRecord, w *UncommittedWriter, pw *writer.ParquetWriter, mark *GCUncommittedMark, runID string, hasData bool) (*GCUncommittedMark, bool, error) {
```

Comment:
Why do we need this? Isn't the grace period in the client sufficient?

Reply:
deleted :(((((((